graylog2-server/src/main/java/org/graylog2/dashboards/widgets/DashboardWidget.java

```diff
@@ -132,17 +135,21 @@ public static DashboardWidget fromPersisted(MetricRegistry metricRegistry, Searc
         }
 
         TimeRange timeRange;
-        if (rangeType.equals("relative")) {
-            timeRange = new RelativeRange((Integer) timerangeConfig.get("range"));
-        } else if (rangeType.equals("keyword")) {
-            timeRange = new KeywordRange((String) timerangeConfig.get("keyword"));
-        } else if (rangeType.equals("absolute")) {
-            String from = new DateTime(timerangeConfig.get("from"), DateTimeZone.UTC).toString(Tools.ES_DATE_FORMAT);
-            String to = new DateTime(timerangeConfig.get("to"), DateTimeZone.UTC).toString(Tools.ES_DATE_FORMAT);
-
-            timeRange = new AbsoluteRange(from, to);
-        } else {
-            throw new InvalidRangeParametersException("range_type not recognized");
+        switch (rangeType) {
```

Comment:
As a lazy-ass developer I like the IntelliJ IDEA automatic refactorings…
!(Yaaaaawn)[http://www.adelgazarysalud.com/wp-content/uploads/2014/07/cute-animal-gifs.gif]